cpp/src/arrow/compute/api.h

```diff
@@ -38,24 +38,16 @@
 /// @{
 /// @}
 
-#include "arrow/compute/exec/expression.h"  // IWYU pragma: export
-
-/// \defgroup execnode-options Concrete option classes for ExecNode options
-/// @{
-/// @}
-
-#include "arrow/compute/exec/options.h"  // IWYU pragma: export
+#include "arrow/compute/expression.h"  // IWYU pragma: export
 
 /// \defgroup execnode-row Utilities for working with data in a row-major format
 /// @{
 /// @}
 
 #include "arrow/compute/row/grouper.h"  // IWYU pragma: export
 
-/// \defgroup execnode-components Components associated with ExecNode
+/// \defgroup execnode-components Components associated with ExecBatch
 /// @{
 /// @}
 
-#include "arrow/compute/exec.h"            // IWYU pragma: export
-#include "arrow/compute/exec/exec_plan.h"  // IWYU pragma: export
-#include "arrow/compute/exec/groupby.h"    // IWYU pragma: export
+#include "arrow/compute/exec.h"  // IWYU pragma: export
```

Comment:
This could probably be split someday (no need to do it in this PR).  `ExecContext` / `ExecSpan` / `ExecResult` / `ExecValue` / `CallFunction` / `GetFunctionExecutor` belong in compute.  `ExecBatch` belongs in `exec`.  `SelectionVector` could probably be removed at this point and added back in later if we ever decide to do something with it.

Reply:
I'll fix the other things in this PR and then make another one to move ExecBatch